x/tokenfactory/keeper/createdenom.go

```diff
@@ -3,61 +3,57 @@ package keeper
 import (
 	"fmt"
 
-	"cosmossdk.io/errors"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 
 	"github.com/neutron-org/neutron/x/tokenfactory/types"
 )
 
-// CreateDenom creates new denom with `factory/{creatorAddr}/{subdenom}` name.
-// Charges creatorAddr fee for creation
-func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr, subdenom string) (newTokenDenom string, err error) {
-	err = k.chargeFeeForDenomCreation(ctx, creatorAddr)
-	if err != nil {
-		return "", errors.Wrapf(types.ErrUnableToCharge, "denom fee collection error: %v", err)
-	}
-
+// ConvertToBaseToken converts a fee amount in a whitelisted fee token to the base fee token amount
+func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr string, subdenom string) (newTokenDenom string, err error) {
 	denom, err := k.validateCreateDenom(ctx, creatorAddr, subdenom)
 	if err != nil {
-		return "", errors.Wrapf(types.ErrInvalidDenom, "denom validation error: %v", err)
+		return "", err
 	}
 
-	err = k.createDenomAfterValidation(ctx, creatorAddr, denom)
+	err = k.chargeForCreateDenom(ctx, creatorAddr)
```

Comment:
As I understand, this is a place we can use the second param (as here https://github.com/osmosis-labs/osmosis/blob/1d9a195c89d07a0e1bba5793b4627db0de7b23a5/x/tokenfactory/keeper/createdenom.go#L87).

So here I see two possible paths:
- Remove `DenomCreationFee` from params (if it was discussed with @pr0n00gler)
- Get back charging logic for `DenomCreationFee`, but send the tokens to the `FeeCollectorAddress` (as in here https://github.com/neutron-org/neutron/blob/be18760ab887dc6df51bba08fe2c27f11a9aca8d/x/feeburner/keeper/keeper.go#L143)